plugins/mobile-plugin/src/main/java/ru/sbtqa/tag/pagefactory/mobile/utils/SwipeUtils.java

```diff
@@ -147,14 +153,14 @@ public static void swipeToText(DirectionStrategy direction, String text, MatchSt
      */
     public static void swipeToText(DirectionStrategy direction, String text, MatchStrategy strategy, int depth) throws SwipeException {
         for (int depthCounter = 0; depthCounter < depth; depthCounter++) {
-            String oldPageSource = PageFactory.getDriver().getPageSource();
+            String oldPageSource = appiumDriver.getPageSource();
             switch (strategy) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a default case to this switch. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this "switch" statement by "if" statements to increase readability. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1301)